meshtastic/node.py

```
@@ -998,27 +1060,112 @@ def setURL(self, url: str, addOnly: bool = False) -> None:
 
             for ignored_name in ignored_channel_names:
                 logger.info(
                     f'Ignoring existing or empty channel "{ignored_name}" from add URL'
                 )
-            for channel_index, channel_name in channels_to_write:
-                logger.info(f"Adding new channel '{channel_name}' to device")
-                self.writeChannel(channel_index)
+            written_indices: list[int] = []
+            try:
+                for channel_index, channel_name in channels_to_write:
+                    logger.info(f"Adding new channel '{channel_name}' to device")
```

Ruff (0.15.5)

[warning] Logging statement uses f-string

(G004)